RPi, Generic: backport proposed patch to reduce IR latency

HiassofT opened the pull request:
See https://www.spinics.net/lists/linux-media/msg131010.html

The current IR decoding is much too slow. Many IR protocols rely on
a trailing space for decoding (e.g. rc-6 needs to know when the bits
end). The trailing space is generated by the IR timeout, and if this
is longer than required, keys can be perceived as sticky and slugish.

The other issue the keyup timer. IR has no concept of a keyup message,
this is implied by the absence of IR. So, minimising the timeout for
this further improves the handling.

With these patches in place, using IR with the builtin decoders is much
improved and feels very snappy.

MilhouseVH commented:
This is in my nightly test builds since 28 March, and no problems reported so far.

On my NUC with Harmony One/RC6 remote these commits are working just fine.

HiassofT commented:
I've updated the backports with v2 of the patch series (see here https://www.spinics.net/lists/linux-media/msg131804.html) plus 2 additional fixes for the mce_kbd decoder (see here https://www.spinics.net/lists/linux-media/msg132145.html)

Changes since v1:
- configurable timeout on mceusb receivers, should make these more snappy as well
- fixes for MCE keyboard decoder

HiassofT commented:
Issues with mceusb receiver should hopefully be fixed now with upstream patch,

I've added back the mceusb commit plus a backport of the fix from here: https://patchwork.linuxtv.org/patch/48782/

HiassofT commented:
I've added a patch that should prevent a race between the keyup timer and incoming scancodes which can lead to bogus key up+down events while a button is pressed

HiassofT commented:
I've dropped today's change, my initial analysis was not correct and it probably doesn't fix the issue reported on the forum

HiassofT commented:
I've added backports of the IR_dprintk()->dev_dbg() changes in upstream 4.17-rc, together with dynamic debug this should allow us to diagnose the mceusb issue reported on the forum

HiassofT commented:
I've dropped the 2 recently added debugging changes (2 milhouse builds with these are enough) and only added back the missing backport change to ir-lirc-codec.

This should be fine now, so I'd say wait a couple more days if issues are reported, then I'll squash it and we can merge

HiassofT commented:
rebased and squashed

The patches have been applied to the upstream media tree and no issues have been reported in the last week. So this should be good to go in
